src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -145,8 +145,8 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
         $defaults = array_replace($globals['defaults'], $annot->getDefaults());
         $requirements = array_replace($globals['requirements'], $annot->getRequirements());
         $options = array_replace($globals['options'], $annot->getOptions());
-        $schemes = array_merge($globals['schemes'], $annot->getSchemes());
-        $methods = array_merge($globals['methods'], $annot->getMethods());
+        $schemes = array_unique(array_merge($globals['schemes'], $annot->getSchemes()));
```

Comment:
The more consistent solution is to reset the globals in line 127 where this is already done for some properties. Actually all properties should be reset, because the globals == $annotation properties in this case.
So I would suggest to extract https://github.com/symfony/symfony/blob/eebcc9d2de715641b6153860de5bff74b9ba1697/src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php#L257-L268 into a new private `resetGlobals()` function that is then called in `getGlobals` and in line 127.